src/plugins/data_source_management/public/components/direct_query_data_sources_components/acceleration_management/acceleration_details_flyout.tsx

```diff
@@ -231,8 +232,9 @@ export const AccelerationDetailsFlyout = (props: AccelerationDetailsFlyoutProps)
   const DiscoverIcon = () => {
     return (
       <EuiButtonEmpty
+        isDisabled={!getUiSettings().get('query:enhancements:enabled')}
```

Comment:
nit: `data_source_management` doesn't take a dependency on the `data` plugin, but could maybe add it [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/common/index.ts) since we also have `DEFAULT_DATA_SOURCE_UI_SETTINGS_ID`.

Reply:
Would there be any benefit to a dependency on the data plugin if we can just check the ui settings? Or were you pointing me to adding this UI setting as a constant to that folder?